debezium-connector-cassandra/src/main/java/io/debezium/connector/cassandra/transforms/type/deserializer/MapTypeDeserializer.java

```diff
@@ -34,4 +36,20 @@ public SchemaBuilder getSchemaBuilder(AbstractType<?> abstractType) {
         Schema valuesSchema = CassandraTypeDeserializer.getSchemaBuilder(valuesType).build();
         return SchemaBuilder.map(keySchema, valuesSchema);
     }
+
+    @Override
+    public Object deserialize(AbstractType<?> abstractType, ComplexColumnData ccd) {
+        List<ByteBuffer> bbList = ((MapType) abstractType).serializedValues(ccd.iterator());
+        AbstractType keyType = ((MapType) abstractType).getKeysType();
+        AbstractType valueType = ((MapType) abstractType).getValuesType();
+        Map<Object, Object> deserializedMap = new HashMap<>();
+        int i = 0;
+        while (i < bbList.size()) {
+            ByteBuffer kbb = bbList.get(i);
+            ByteBuffer vbb = bbList.get(i + 1);
```

Comment:
Ok, that scared me at first, but it seems it's guaranteed that the list size is always a multiple of 2?

Reply:
Yes since for each cell in a map-type column, cassandra add the ByteBuffer of both key and value into the ByteBuffer list: https://www.codota.com/web/assistant/code/rs/5c655eac968485000193d273#L220

Reply:
> Ok, that scared me at first, but it seems it's guaranteed that the list size is always a multiple of 2?

Yes the size should always be even since for each cell in a map-type column, cassandra add the ByteBuffer of both key and value into the ByteBuffer list: https://www.codota.com/web/assistant/code/rs/5c655eac968485000193d273#L220